fix: increment seqno before execute calls to prevent InvalidArgument …

larkee opened the pull request:
…errors after a previous error

Fixes #11

tseaver commented on google/cloud/spanner_v1/transaction.py:
```diff
@@ -191,18 +191,19 @@ def execute_update(self, dml, params=None, param_types=None, query_mode=None):
         api = database.spanner_api
 
+        seqno = self._execute_sql_count
+        self._execute_sql_count += 1
```
Multiple assignment would be atomic here, e.g.:
```suggestion
        seqno, self._execute_sql_count = self._execute_sql_count, self._execute_sql_count + 1
```

tseaver commented on google/cloud/spanner_v1/transaction.py:
```diff
@@ -243,15 +244,16 @@ def batch_update(self, statements):
         api = database.spanner_api
 
+        seqno = self._execute_sql_count
+        self._execute_sql_count += 1
```
Likewise here:
```
        seqno, self._execute_sql_count = self._execute_sql_count, self._execute_sql_count + 1
```

tseaver commented:
@larkee I don't know what the CLA bot is on about:  `tseaver@palladion.com` is my normal github email, and the one associated with my long-ago-signed CLA (I've been committing to these repos using it for 5+ years now!).

tseaver commented:
Also, the PR still needs the added unit tests I outlined above.

larkee commented:
@tseaver I've fixed the CLA problem by just redoing the commits. I added the test and noticed that the change to snapshot.py doesn't actually make any difference, possibly because the call is wrapped in functools.partial? Because of this, I've reverted it.

odeke-em commented:
Kindly pinging you @tseaver @skuruppu to take a look here as we need this change. Thank you.
